courses/machine_learning/deepdive2/structured/solutions/4a_sample_babyweight.ipynb

```diff
@@ -0,0 +1,2808 @@
+{
```

Comment:
<p>Since orderby is used only once, why not just pass the full query? You could even take advantage of bigquery's parameterized queries feature and set the limit equal the df.head count since there's not point in pulling in more rows than what is being displayed.</p><p> </p><p><a href="https://cloud.google.com/bigquery/docs/parameterized-queries" target="_blank">https://cloud.google.com/bigquery/docs/parameterized-queries</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/671/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
<p>In case they want to use it for other things that involve order bys. I made the limit the same as count, but didn't want to overly complicate the code with extra stuff as to get too far away from the simple idea.</p>

Reply:
Running the full query would simplify it, wouldn't it? Right now, we're putting the query into a subquery which is SELECT * FROM (passed_query), so it seems like we're adding an extra level for not much value.

Reply:
<p>This is the standard way we've done it in our other labs where we are adding things before and/or after another query. Therefore, this is a simple way for students to understand both the base query and how we are modifying it to form our modified query depending on what parameters we pass in.</p>

Reply:
Consider this, Ryan. Would you say that 
SELECT * FROM (SELECT * FROM my_table) LIMIT 10
is optimal and readable code?

Reply:
Discussed in our team meeting. As Dan and Chris pointed out this is best for error proofing if someone wants to add to the query and most importantly this is consistent with previous labs throughout the 10 courses where we nest a query within another. Seems more of a nit all of us on the call.